rayon-core/src/scope/mod.rs

```diff
@@ -381,17 +394,70 @@ where
 {
     in_worker(|owner_thread, _| {
         let scope = ScopeFifo::<'scope>::new(owner_thread);
-        unsafe { scope.base.complete(owner_thread, || op(&scope)) }
+        unsafe { scope.base.complete(Some(owner_thread), || op(&scope)) }
     })
 }
 
-impl<'scope> Scope<'scope> {
-    fn new(owner_thread: &WorkerThread) -> Self {
-        Scope {
-            base: ScopeBase::new(owner_thread),
+/// Creates a "fork-join" scope `s` and invokes the closure with a
+/// reference to `s`. This closure can then spawn asynchronous tasks
+/// into `s`. Those tasks may run asynchronously with respect to the
+/// closure; they may themselves spawn additional tasks into `s`. When
+/// the closure returns, it will block until all tasks that have been
+/// spawned into `s` complete.
+///
+/// This is just like `scope()` except the closure runs on the same thread
+/// that calls `external_scope()`. Only work that it spawns runs in the
```

Comment:
Only calls to spawn()? Would be good to have the fifo variant at the same time?

Reply:
How about we get this reviewed and landed first so I don't duplicate any mistakes I've made?

Reply:
Sounds good to me.